planet/src/main/java/com/google/planet/servlets/UserServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.planet.servlets;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.ValueEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+@WebServlet("/user/*")
+public class UserServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    final String username = request.getPathInfo().substring(1);
+    UserServlet.initializeFirebase();
+    final FirebaseDatabase database = FirebaseDatabase.getInstance();
+    final DatabaseReference reference = database.getReference("/users");
+    reference.orderByChild("username").equalTo(username).addListenerForSingleValueEvent(
+    new ValueEventListener() {
+      @Override
+      public void onDataChange(DataSnapshot dataSnapshot) {
+        if (!dataSnapshot.exists()) {
+          System.out.println("Profile Not Found");
+          return;
+        }
+      }
+      
+      @Override
+      public void onCancelled(DatabaseError databaseError) {
+        // This method has to be overrided that's why I'm adding it
```

Comment:
How/if this method gets triggered? When it does, what will the user sees?

Reply:
"This method will be triggered in the event that this listener either failed at the server, or is removed as a result of the security and Firebase Database rules." I will print something when this method gets triggered!